TheJumpCloud/support
# SA-3040: Add/Remove-JCCloudDirectory Member Functions

## Issues
* [SA-3040](https://jumpcloud.atlassian.net/browse/SA-3040) - Add/Remove-JCCloudDirectory Member Functions

## What does this solve?
This introduces the following functions:
* Add-JCOffice365Member
* Add-JCGsuiteMember
* Remove-JCOffice365Member
* Remove-JCGsuiteMember

Currently, admins need to utilize the Add/Remove-JCAssociation functions to adjust their memberships to these groups. With the introduction of the Get/Set-JCCloudDirectory functions, it would make it easier on the admins to programatically adjust their CloudDirectory memberships via easy and intuitive functions.

## Is there anything particularly tricky?
N/A

## How should this be tested?
Pester test suites have been created for each of the functions. However we should test to ensure that correct information is being passed into the function, example:

1. `Add-JCOffice365Member -ID <Office365InstanceID> -Username John.Doe -UserID <userID>` | This should throw, only one property should be passed per call (either username OR userid, not both)
2. `Add-JCOffice365Member -ID <Office365InstanceID> -GroupName Office365Users -GroupID <groupID>` | This should throw, only one property should be passed per call (either groupName OR groupID, not both)
3. `Add-JCOffice365Member -ID <Office365InstanceID> -Username John.Doe -GroupName 'Office365Users'` | This should throw. Cannot specify both user and usergroup at the same time

These 3 exceptions should be present for all 4 functions
